docs/WatchPlugins.md

```diff
@@ -148,3 +148,24 @@ class MyWatchPlugin {
   }
 }
 ```
+
+**Note**: If you do call `updateConfigAndRun`, your `run` method should not resolve to a truthy value, as that would trigger a double-run.
+
+#### Authorized configuration keys
+
+For stability and safety reasons, only part of the global configuration keys can be updated with `updateConfigAndRun`. The current white list is as follows:
```

Comment:
https://facebook.github.io/jest/docs/en/cli.html#changedsince, https://facebook.github.io/jest/docs/en/cli.html#lastcommit and https://facebook.github.io/jest/docs/en/cli.html#onlychanged would be awesome as well. Maybe https://facebook.github.io/jest/docs/en/cli.html#logheapusage?

Reply:
Hey @SimenB!

- `--changedSince` and `--lastCommit` do not have any direct equivalent in the global configuration object. They're used by `getChangedFilesForRoots` in `jest-changed-files`, and are also ignored when `--onlyChanged` (which is the default watch mode), so they don't seem to make sense in that mode.  At any rate, the watch architecture would not let us change that on the fly right now, this is a bootstrap-time computation.
- `--onlyChanged` is already togglable in watch mode: these are the `a` key (all tests, same as `--watchAll`) vs. `o` key (only changed tests, same as `--onlyChanged`).
- `--logHeapUsage` does have a matching boolean in the global config, so I'm open to adding it, what do others think?

Reply:
in the same way toggling `onlyChanged` is useful, I think `changedSince` would be useful. It would then overwrite `onlyChanged`.

Reply:
@SimenB I agree it would be nice, but Jest's current architecture doesn't let us do it. It computes the watched list at boot time, then keeps it in a closure where specific global config options may amend it (such as the watch mode). There's no stuff in there right now for `changedSince` or `lastCommit`, and I feel the change would be significant.

This specific PR is bound by what's in the global config. Should this feature set become config-based, I'll be glad to expand the white list, but it's a different feature scope IMHO.

Reply:
@rogeliog sounds like we need another of your magical refactor runs 😀

Reply:
Yeah we would need to change a couple of things in order to make that happen 🙃